### ISSUE TITLE
Unpopulated relations are included in serialization

### ISSUE THREAD

**Issue by lamkashingpaul** [mikro-orm/mikro-orm#6194]
### Describe the bug

Relations that are not populated are included in the serialization. I have checked some existing issues, such as #4138 and #4111, and found that the serialization works properly when objects are fetched using the entity manager.

I am fetching and retrieving my objects using the query builder and joining subqueries since I have relatively complicated conditions for populating the relations. In my case, nested relations that are not populated are serialized.

I tried to manually map and populate the relations after fetching objects with the query builder (as implemented in the last test of the reproduction), and it seemed to resolve the issue. However, I am not sure if this is the correct way to do it.

### Reproduction

https://github.com/lamkashingpaul/reproduction

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.13

### Node.js version

22.9.0

### Operating system

Windows

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Thanks for the repro, looks valid.

I wonder if we should refactor the internals to use explicit serialization, which won't suffer from this (passing the implicit serialization hint to it should result in the same behavior). We could only keep the current implicit serialization logic for when there is no populate hint stored on the root entity.

**Comment by B4nan** [comment]
Btw looking closer at the test case, I feel like you don't understand what `em.create` is actually capable of, this works too, no need to create some temporary shop entities just to accommodate the strict type safety, or call it 4 times, providing the type parameters, or using the `ref` helper:

```ts
orm.em.create(User, {
  id: 1,
  name: 'User 1',
  shop: {
    id: 1,
    name: 'Shop 1',
    orders: [
      {
        id: 1,
        time: new Date('2024-10-24T01:00:00Z'),
      },
      {
        id: 2,
        time: new Date('2024-10-24T02:00:00Z'),
      },
    ],
  },
});
```

Same applies to `em.assign`.

**Comment by B4nan** [comment]
Those one-liners, you gotta love them. The `raw` support in `qb.distinctOn` is fixed via f8580c8c74e8d8677dfb37b67689d3a0e31a2630 and the serialization via 3850c1e2b2581ea205c39e80171acb039f10561d
